File: src/main/java/team4/footwithme/resevation/service/ParticipantService.java
package team4.footwithme.resevation.service;

import team4.footwithme.member.domain.Member;
import team4.footwithme.resevation.service.request.ParticipantUpdateServiceRequest;
import team4.footwithme.resevation.service.response.ParticipantResponse;

import java.util.List;

public interface ParticipantService {
    ParticipantResponse createMercenaryParticipant(Long mercenaryId, Member member);

    ParticipantResponse createParticipant(Long reservationId, Member member);

    Long deleteParticipant(Long reservationId, Member member);

    ParticipantResponse updateMercenaryParticipant(ParticipantUpdateServiceRequest request, Member member);

    List<ParticipantResponse> getAcceptParticipants(Long reservationId);

    List<ParticipantResponse> getParticipants(Long reservationId);

    List<ParticipantResponse> getParticipantsMercenary(Long reservationId);
}

File: src/main/java/team4/footwithme/resevation/service/ParticipantService.kt
package team4.footwithme.resevation.service

import team4.footwithme.member.domain.Member
import team4.footwithme.resevation.service.request.ParticipantUpdateServiceRequest
import team4.footwithme.resevation.service.response.ParticipantResponse

interface ParticipantService {
    fun createMercenaryParticipant(mercenaryId: Long?, member: Member?): ParticipantResponse

    fun createParticipant(reservationId: Long?, member: Member?): ParticipantResponse

    fun deleteParticipant(reservationId: Long, member: Member?): Long?

    fun updateMercenaryParticipant(request: ParticipantUpdateServiceRequest?, member: Member?): ParticipantResponse

    fun getAcceptParticipants(reservationId: Long?): List<ParticipantResponse>?

    fun getParticipants(reservationId: Long?): List<ParticipantResponse>?

    fun getParticipantsMercenary(reservationId: Long?): List<ParticipantResponse>?
}
